From db5a5e67f7715e4087ae2a9685fdf4ff0c4fca17 Mon Sep 17 00:00:00 2001
From: Zdenek Kabelac <zkabelac@redhat.com>
Date: Tue, 5 Jun 2012 11:26:54 +0200
Subject: [PATCH 3/3] Sync filesystem for thin snapshots

Add missing lockfs option when suspend origin, before thin volume
snapshot is created

(cherry picked from commit 2f99e5e35a9918c337e66b74aa6126cca9328c64)

Conflicts:
	WHATS_NEW
---
 WHATS_NEW               |    1 +
 lib/activate/activate.c |    5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

--- a/WHATS_NEW
+++ b/WHATS_NEW
@@ -1,6 +1,7 @@
 Version 2.02.95-wrs
 ================================
   Fix lvrename for thin volumes (regression in for_each_sub_lv() 2.02.89).
+  Fix missing sync of filesystem when creating thin volume snapshot.
 
 Version 2.02.95 - 6th March 2012
 ================================
--- a/lib/activate/activate.c
+++ b/lib/activate/activate.c
@@ -1488,6 +1488,9 @@ static int _lv_suspend(struct cmd_contex
 	    (lv_is_origin(lv_pre) || lv_is_cow(lv_pre)))
 		lockfs = 1;
 
+	if (laopts->origin_only && lv_is_thin_volume(lv) && lv_is_thin_volume(lv_pre))
+		lockfs = 1;
+
 	/*
 	 * Suspending an LV directly above a PVMOVE LV also
  	 * suspends other LVs using that same PVMOVE LV.
@@ -1567,7 +1570,7 @@ static int _lv_resume(struct cmd_context
 	if (lv_is_thin_pool(lv) && laopts->origin_only)
 		messages_only = 1;
 
-	if (!lv_is_origin(lv))
+	if (!lv_is_origin(lv) && !lv_is_thin_volume(lv))
 		laopts->origin_only = 0;
 
 	if (test_mode()) {
